src/Microsoft.PowerShell.Commands.Utility/commands/utility/group-object.cs

```diff
@@ -368,7 +368,7 @@ protected override void EndProcessing()
                             {
                                 if (_grp.Values.Count == 1)
                                 {
-                                    _table.Add(_grp.Values[0], _grp.Group);
+                                    _table.Add(PSObject.Base(_grp.Values[0]), _grp.Group);
```

Comment:
I suggest leaving a comment here since the change is rather subtle; otherwise, the change looks good to me.

Reply:
Yes, the comment would be useful.